pkg/platform/kube/client/deployer.go

```diff
@@ -146,6 +159,215 @@ func (d *Deployer) Deploy(ctx context.Context,
+	}
+
+	// if secret exists, get the data
+	if functionSecretData != nil {
+		functionSecretMap, err := functionconfig.DecodeSecretData(functionSecretData)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to decode function secret data")
+		}
+		return functionSecretMap, nil
+	}
+
+	// secret doesn't exist
+	d.logger.DebugWithCtx(ctx, "Function secret doesn't exist", "functionName", functionName, "functionNamespace", functionNamespace)
+	return nil, nil
+}
+
+func (d *Deployer) getFunctionSecretData(ctx context.Context, functionName, functionNamespace string) (map[string][]byte, error) {
+
+	// get existing function secret
+	secretName := d.generateFunctionSecretName(functionName)
+	functionSecret, err := d.consumer.KubeClientSet.CoreV1().Secrets(functionNamespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, errors.Wrap(err, "Failed to get function secret")
+		}
+		return nil, nil
+	}
+	return functionSecret.Data, nil
+}
+
+func (d *Deployer) createOrUpdateFunctionSecret(ctx context.Context,
+	encodedSecretsMap map[string]string,
+	name,
+	namespace string) error {
+
+	secretConfig := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: d.generateFunctionSecretName(name),
+			Labels: map[string]string{
+				"nuclio.io/function-name": name,
+			},
+		},
+		Type:       functionconfig.NuclioSecretType,
+		StringData: encodedSecretsMap,
+	}
+
+	if len(encodedSecretsMap) > 0 {
+		if err := d.createOrUpdateSecret(ctx, namespace, secretConfig); err != nil {
+			return errors.Wrap(err, "Failed to create function secret")
+		}
+		return nil
+	}
+
+	// if secret exists and there are no secrets to set, delete the secret
+	return d.deleteExistingSecret(ctx, namespace, secretConfig.Name)
+}
+
+func (d *Deployer) generateFunctionSecretName(functionName string) string {
+	return fmt.Sprintf("%s%s", functionconfig.NuclioSecretNamePrefix, functionName)
+}
+
+func (d *Deployer) handleFlexVolumeSecret(ctx context.Context, functionName, functionNamespace string, secretsMap map[string]string) error {
+
+	flexVolumeSecretName := fmt.Sprintf("%s%s", functionconfig.NuclioFlexVolumeSecretNamePrefix, functionName)
+
+	// check if flex volume secret exists
+	for secretKey, secretValue := range secretsMap {
+		if strings.Contains(strings.ToLower(secretKey), "flexvolume") &&
+			strings.Contains(strings.ToLower(secretKey), "accesskey") {
+			d.logger.DebugWithCtx(ctx, "Found flex volume secret")
+
+			// create or update flex volume secret
+			if err := d.createOrUpdateSecret(ctx, functionNamespace, &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: flexVolumeSecretName,
+					Labels: map[string]string{
+						"nuclio.io/function-name": functionName,
+					},
+				},
+				Type: "v3io/fuse",
+				StringData: map[string]string{
+					"accessKey": secretValue,
+				},
+			}); err != nil {
+				return errors.Wrap(err, "Failed to create or update flex volume secret")
```

Comment:
thinking about it again, the issue is tht you essentially limiting the function to support a single access key

Reply:
A single access key only for the flex volume, yes.
But in the `FlexVolume.Options` map you can only have one `accessKey` key anyways.